### PR TITLE
iosxr: Added new states to lldp resource module

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY
partially resolve : #16

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
iosxr_lldp_global.py
iosxr_lldp_interfaces.py

### CODE REVIEW

**Comment by NilashishC** [comment]
Test for lldp_global is red

File: plugins/module_utils/network/iosxr/config/lldp_global/lldp_global.py [hunk 1]
```diff
@@ -69,20 +69,43 @@ def execute_module(self):
...
 
-        result["before"] = existing_lldp_global_facts
-        if result["changed"]:
-            result["after"] = changed_lldp_global_facts
+        if self.state in self.ACTION_STATES:
+            result["before"] = existing_lldp_global_facts
+            if result["changed"]:
+                result["after"] = changed_lldp_global_facts
+        elif self.state == "gathered":
+            result["gathered"] = changed_lldp_global_facts
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/iosxr/config/lldp_global/lldp_global.py:108]: in set_config(), IMO not want check is not required anymore.

File: plugins/module_utils/network/iosxr/config/lldp_interfaces/lldp_interfaces.py [hunk 2]
```diff
@@ -74,19 +74,43 @@ def execute_module(self):
         warnings = list()
         commands = list()
 
-        existing_lldp_interfaces_facts = self.get_lldp_interfaces_facts()
-        commands.extend(self.set_config(existing_lldp_interfaces_facts))
-        if commands:
+        if self.state in self.ACTION_STATES:
+            existing_lldp_interfaces_facts = self.get_lldp_interfaces_facts()
+        else:
+            existing_lldp_interfaces_facts = {}
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/iosxr/config/lldp_interfaces/lldp_interfaces.py:80]: this should be a list . right?

File: plugins/modules/iosxr_lldp_global.py [hunk 3]
```diff
@@ -37,11 +37,11 @@
 }
 
 DOCUMENTATION = """module: iosxr_lldp_global
-short_description: Manage Global Link Layer Discovery Protocol (LLDP) settings on
-  IOS-XR devices.
+short_description: LACP global resource module.
```
Comment (rohitthakur2590) [inline plugins/modules/iosxr_lldp_global.py:40]: replace LACP with LLDP

File: plugins/modules/iosxr_lldp_interfaces.py [hunk 4]
```diff
@@ -36,12 +36,12 @@
     "supported_by": "network",
 }
 
-DOCUMENTATION = """module: iosxr_lldp_interfaces
-short_description: Manage Link Layer Discovery Protocol (LLDP) attributes of interfaces
-  on IOS-XR devices.
+DOCUMENTATION = """module: cisco.iosxr.iosxr_lldp_interfaces
+short_description: Lldp interfaces resource module.
```
Comment (rohitthakur2590) [inline plugins/modules/iosxr_lldp_interfaces.py:40]: could be LLDP interfaces resource module.

File: plugins/modules/iosxr_lldp_interfaces.py [hunk 5]
```diff
@@ -36,12 +36,12 @@
     "supported_by": "network",
 }
 
-DOCUMENTATION = """module: iosxr_lldp_interfaces
-short_description: Manage Link Layer Discovery Protocol (LLDP) attributes of interfaces
-  on IOS-XR devices.
+DOCUMENTATION = """module: cisco.iosxr.iosxr_lldp_interfaces
```
Comment (rohitthakur2590) [inline plugins/modules/iosxr_lldp_interfaces.py:39]: would this be cisco.iosxr.iosxr_lldp_interfaces of just iosxr_lldp_interfaces  ?

**Comment by GomathiselviS** [comment]
recheck

File: plugins/module_utils/network/iosxr/config/lldp_global/lldp_global.py [hunk 6]
```diff
@@ -112,7 +135,7 @@ def set_state(self, want, have):
                   to the desired configuration
         """
         state = self._module.params["state"]
-        if state in ("merged", "replaced") and not want:
+        if state in ("merged", "replaced", "rendered") and not want:
```
Comment (NilashishC) [inline plugins/module_utils/network/iosxr/config/lldp_global/lldp_global.py:138]:
```suggestion
        if state in ("merged", "replaced", "rendered", "overridden") and not want:
```

Comment (GomathiselviS) [reply plugins/module_utils/network/iosxr/config/lldp_global/lldp_global.py:138]: This module does not have overridden state

**Pull request merged**